test/components/views/settings/devices/DeviceDetailHeading-test.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { fireEvent, render, RenderResult } from '@testing-library/react';
+
+import { DeviceDetailHeading } from '../../../../../src/components/views/settings/devices/DeviceDetailHeading';
+import { flushPromisesWithFakeTimers } from '../../../../test-utils';
+
+jest.useFakeTimers();
+
+describe('<DeviceDetailHeading />', () => {
+    const device = {
+        device_id: '123',
+        display_name: 'My device',
+        isVerified: true,
+    };
+    const defaultProps = {
+        device,
+        saveDeviceName: jest.fn(),
+    };
+    const getComponent = (props = {}) =>
+        <DeviceDetailHeading {...defaultProps} {...props} />;
+
+    const setInputValue = (getByTestId: RenderResult['getByTestId'], value: string) => {
+        const input = getByTestId('device-rename-input');
+
+        fireEvent.change(input, { target: { value } });
+    };
+
+    it('renders device name', () => {
+        const { container } = render(getComponent());
+        expect({ container }).toMatchSnapshot();
```

Comment:
To check if it renders the device name we could test for it.
Scalpel vs snapshot-shotgun :laughing: